controllers/user.controller.js

```diff
@@ -1,84 +1,129 @@
 //Controller for user routes
-const {validationResult} = require('express-validator')
+const { validationResult } = require("express-validator");
 
 //Import the user model
-const User = require('../models/user.model')
+const User = require("../models/user.model");
 
-exports.createUser = (req, res, next) =>
-{
-  const errors = validationResult(req)
-  if(!errors.isEmpty())
-  {
-    const error = new Error('Validation failed')
-    error.httpStatus = 422
-    error.data = errors.array()
-    throw error
+exports.createUser = (req, res, next) => {
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    const error = new Error("Validation failed");
+    error.httpStatus = 422;
+    error.data = errors.array();
+    throw error;
   }
-  const username = req.body.username
-  const password = req.body.password
-  const name = req.body.name
-  const role = req.body.role
-  const email = req.body.email
-  const accountStatus = req.body.accountStatus
+  const username = req.body.username;
+  const password = req.body.password;
+  const name = req.body.name;
+  const role = req.body.role;
+  const email = req.body.email;
+  const accountStatus = req.body.accountStatus;
 
   const user = new User({
     username,
     password,
     name,
     email,
     role,
-    accountStatus
-  })
+    accountStatus,
+  });
 
   user
-  .save()
-  .then(result =>
-    {
-      res.status(200)
-      .json({
-        message: 'User created successfully!', 
-        userId: result._id
-      })
+    .save()
+    .then((result) => {
+      res.status(200).json({
+        message: "User created successfully!",
+        userId: result._id,
+      });
     })
-  .catch(err =>
-    {
-      const error = new Error('User creation error!')
-      error.message = err.message
-      error.data = err.errors
-      next(error)
+    .catch((err) => {
+      const error = new Error("User creation error!");
+      error.message = err.message;
+      error.data = err.errors;
+      next(error);
+    });
+};
+
+//Return List of Users
+exports.getUsers = (req, res, next) => {
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    const error = new Error("Validation failed");
+    error.httpStatus = 422;
+    error.data = errors.array();
+    throw error;
+  }
+
+  User.find()
+    .then((users) => {
+      if (!users) {
+        const error = new Error("No users found!");
+        error.httpStatus = 404;
+        throw error;
+      }
+      return res.status(200).json(users);
```

Comment:
For security reasons, we might not want to send back the whole document for a user with this call. We should probably map the necessary values that should be sent back.